test/con_cache_test.exs

```diff
@@ -184,6 +184,24 @@ defmodule ConCacheTest do
     end)
   end
 
+  test "get_or_store on bag" do
+    with_cache([ets_options: [:bag]],fn(cache) ->
+      assert ConCache.get_or_store(cache, :a, fn() -> 1 end) == 1
```

Comment:
This is a bit weird, because we can insert only one value, and on first store we fetch that value, while on next one we fetch a list. 

I think we should support something like `ConCache.get_or_store(cache, :a, fn() -> [1, 2, 3] end) == [1, 2, 3]`, because it feels more consistent.

Glancing at the code, I think the change might be a bit tricky, so I can try to do this myself. If you're still willing to take a stab at it, let me know. Otherwise, I'll merge your changes and try it myself.